### PR TITLE
Calibration dataset size < `batch` fix

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🛠️ This PR makes documentation builds stricter and improves INT8 export calibration by handling small datasets more gracefully instead of failing unnecessarily.

### 📊 Key Changes
- 📚 Updated the docs build command in `docs/build_docs.py` to use `--strict` with `zensical build`.
- ✅ This means documentation builds will now fail on warnings or issues that were previously easier to miss.
- 🤖 Improved INT8 calibration dataloader logic in `ultralytics/engine/exporter.py`.
- 🔍 Added a check to ensure the calibration dataset has at least 1 image.
- 📉 If the calibration dataset has fewer images than the requested batch size, the exporter now automatically reduces the batch size instead of raising an error.
- ⚠️ Existing warnings for small calibration datasets remain, including format-specific guidance for Axelera and general INT8 export recommendations.
- 🚚 The dataloader now uses the adjusted batch size when building the calibration loader.

### 🎯 Purpose & Impact
- 🧹 **Better documentation quality:** Strict doc builds help catch broken docs earlier, improving reliability for contributors and users reading the documentation.
- 💡 **More user-friendly exports:** Small calibration datasets no longer cause immediate export failure just because the batch size is too large.
- 🔄 **Improved robustness:** Export workflows can continue in more edge cases, especially for quick tests or limited datasets.
- ⚠️ **Clearer safeguards:** Users still get warnings when calibration datasets are smaller than recommended, helping maintain export quality expectations.
- 🚀 **Smoother developer experience:** This reduces friction in INT8 model export while also tightening standards for documentation maintenance.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you shortly. Please review the checklist below for smooth integration ✨

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/engine/exporter.py [hunk 1]
```diff
@@ -670,16 +670,18 @@ def get_int8_calibration_dataloader(self, prefix=""):
...
         )
         n = len(dataset)
+        if n < 1:
+            raise ValueError(f"The calibration dataset must have at least 1 image, but found {n} images.")
+        batch = min(self.args.batch, n)
         if n < self.args.batch:
-            raise ValueError(
-                f"The calibration dataset ({n} images) must have at least as many images as the batch size "
-                f"('batch={self.args.batch}')."
+            LOGGER.warning(
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/exporter.py:677]: 💡 **MEDIUM**: Changing the first branch from a hard failure to a warning means the existing small-dataset recommendation warnings no longer fire for `n < self.args.batch` because this is now an `if/elif/elif` chain. For example, a non-Axelera calibration set with 16 images and `batch=32` will only log the batch reduction warning and skip the `>300 images recommended` warning, which changes the previous guidance instead of preserving it.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The PR is small and the core fix looks good, but there is one logic regression: the new warning branch suppresses the existing small-calibration recommendation warnings for undersized datasets. Aside from that, the empty-dataset guard and dynamic batch adjustment are sensible improvements.

💬 Posted 1 inline comment

**Comment by UltralyticsAssistant** [comment]
Amazing to see this merged, @glenn-jocher — thank you for another thoughtful improvement to Ultralytics! 🚀

As Benjamin Franklin said, *“An investment in knowledge pays the best interest.”* This PR reflects that perfectly: stricter docs builds help protect the quality of shared knowledge, while the more graceful INT8 calibration handling makes exports more robust and user-friendly in real-world edge cases.

Really appreciate the care here in improving both developer experience and reliability. Contributions like this keep Ultralytics moving forward with clarity, stability, and polish. 🙌

**Comment by glenn-jocher** [comment]
Thanks — credit goes to the Ultralytics team and YOLO community here; this is now merged into `main`, and the updated small-dataset INT8 calibration behavior is documented in the [exporter reference](https://docs.ultralytics.com/reference/engine/exporter).
